apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Query.cs

```diff
@@ -417,6 +419,13 @@ public Query Select(params FieldPath[] fieldPaths)
         public Query WhereNotIn(FieldPath fieldPath, IEnumerable values) =>
             Where(fieldPath, FieldOp.NotIn, ValidateWhereInValues(values));
 
+        /// <summary>
+        /// Returns an <see cref="AggregateQuery"/> with count(*) aggregation applied on this query.
+        /// </summary>
+        /// <returns>An instance of <see cref="AggregateQuery"/> with count(*) aggregation applied.</returns>
+        public AggregateQuery Count() =>
+            new AggregateQuery(this).WithAggregation(new Aggregation { Count = new Count() });
```

Comment:
Right, and this would become `new AggregateQuery(this).WithAggregation(AggregateQuery.CreateCountAggregate))`. Or you might introduce an `Aggregates` static class as a factory for aggregates. (You'd need to expose the count alias in there, of course.)